exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/transactions/ThreadLocalTransactionManager.kt

```diff
@@ -221,6 +248,20 @@ fun <T> inTopLevelTransaction(
                 if (repetitions >= repetitionAttempts) {
                     throw e
                 }
+                // set delay value
+                val delay = if (minRepetitionDelay < maxRepetitionDelay) {
+                    Random(System.currentTimeMillis()).nextLong(minRepetitionDelay, maxRepetitionDelay)
+                } else if (minRepetitionDelay == maxRepetitionDelay) {
+                    minRepetitionDelay
+                } else {
+                    0
+                }
+                exposedLogger.warn("Wait $delay milliseconds before retrying")
+                try {
+                    Thread.sleep(delay)
```

Comment:
Could you tell me how it will work with `newSuspendTransaction`?

Reply:
I tried to do some tests using newSuspendedTransaction with the repetition delay. Here's what I found :
- newSuspendedTransaction calls suspendedTransactionAsyncInternal which does not handle transaction repetitions
```kotlin
try {
        transaction.statement().apply {
            if (shouldCommit) transaction.commit()
        }
    } catch (e: SQLException) {
        handleSQLException(e, transaction, 1)
        throw e
``` 
-  if newSuspendedTransaction is called within a normal transaction, then the transactions are repeated with the calculated delay.

Is there something else I should check?

Reply:
As a user, I would expect that both `transaction` and `newSuspendTransaction` had the same retry policy

Reply:
Sure, let's move it to the separate issue